From 02a362a4b618081934bbd6116d561c11f329120f Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:35:41 -0400
Subject: [block] blk-mq: remove debug BUG_ON() when draining software queues

Message-id: <1402670238-13916-54-git-send-email-snitzer@redhat.com>
Patchwork-id: 83789
O-Subject: [RHEL7.1 PATCH 053/150] blk-mq remove debug BUG_ON() when draining software queues
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit 98bc1f272aba620d4222120853011d0ef026cf56
Author: Jens Axboe <axboe@fb.com>
Date:   Wed Apr 30 13:43:08 2014 -0600

    blk-mq remove debug BUG_ON() when draining software queues

    It's never been of any use, lets get rid of it.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 0aee325..77308a8 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -599,7 +599,6 @@ static void __blk_mq_run_hw_queue(struct blk_mq_hw_ctx *hctx)
  for_each_set_bit(bit, hctx->ctx_map, hctx->nr_ctx) {
   clear_bit(bit, hctx->ctx_map);
   ctx = hctx->ctxs[bit];
-  BUG_ON(bit != ctx->index_hw);
 
   spin_lock(&ctx->lock);
   list_splice_tail_init(&ctx->rq_list, &rq_list);
-- 
1.7.1